network/config.go

```diff
@@ -61,12 +62,34 @@ func DefaultConfig() *Config {
 	}
 }
 
+func validateAddress(address []string) bool {
```

Comment:
`validateAddress` should be `validateAddresses` and the function should return `error`.